src/scripts/main.js

```diff
@@ -1,3 +1,388 @@
+
+    return regex.test(trimValue);
+  }
+
+  function isValidName(value) {
+    if (
+      typeof value === 'string' &&
+      value.trim().length >= 4 &&
+      isNameOrPosition(value)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidPosition(value) {
+    if (
+      typeof value === 'string' &&
+      value.trim().length >= 1 &&
+      isNameOrPosition(value)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidAge(value) {
+    if (!isNaN(Number(value)) && value >= 18 && value <= 90) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidSalary(value) {
+    if (!isNaN(Number(value))) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function numberToDollar(amount) {
+    const thousandString = Number(amount).toLocaleString('en-US');
+
+    const characters = thousandString.split('');
+
+    characters.unshift('$');
+
+    const result = characters.join('');
+
+    return result;
+  }
+
+  if (
+    isValidName(nameInput.value) &&
+    isValidPosition(positionInput.value) &&
+    isValidAge(ageInput.value) &&
+    isValidSalary(salaryInput.value)
+  ) {
+    const newTr = document.createElement('tr');
+
+    tbody.append(newTr);
+
+    const nameTd = document.createElement('td');
+    const positionTd = document.createElement('td');
+    const officeTd = document.createElement('td');
+    const ageTd = document.createElement('td');
+    const salaryTd = document.createElement('td');
+
+    nameTd.textContent = nameInput.value;
+    positionTd.textContent = positionInput.value;
+    officeTd.textContent = officeInput.value;
+    ageTd.textContent = ageInput.value;
+    salaryTd.textContent = numberToDollar(salaryInput.value);
+
+    newTr.append(nameTd);
+    newTr.append(positionTd);
+    newTr.append(officeTd);
+    newTr.append(ageTd);
+    newTr.append(salaryTd);
+
+    pushNotification(10, 10, 'success');
+  } else {
+    pushNotification(10, 10, 'error');
+  }
+}
+
+formButton.addEventListener('click', clickOnButton);
+
+// #endregion add form
+
+// #region notification
+
+const pushNotification = (posTop, posRight, type) => {
+  const newDiv = document.createElement('div');
+  const titleElem = document.createElement('h2');
+  const descriptionELem = document.createElement('p');
+  const style = document.createElement('style');
+
+  style.innerHTML = `
+  .notification {
+    position: fixed;
+    width: 300px;
+    min-height: 100px;
+    padding: 0 16px;
+    box-sizing: border-box;
+    z-index: 2;
+
+    border-radius: 10px;
+    background: rgba(132, 132, 132, 0.3);
+
+    top: 10px;
+    right: 10px;
+
+    .title {
+      display: block;
+      font-size: 20px;
+      font-weight: 900;
+    }
+
+    &.success {
+      background: rgba(10, 189, 0, 0.5);
+    }
+
+    &.error {
+      background: rgba(253, 0, 0, 0.5);
+    }
+  }
+  `;
+
+  document.head.appendChild(style);
```

Comment:
The `pushNotification` function appends a new style element to the document head each time it's called. This could lead to performance issues if the function is called frequently. Consider moving the style element creation outside the function or checking if it already exists before appending.